common/src/test/java/io/pravega/common/concurrent/FuturesTests.java

```diff
@@ -21,23 +24,40 @@
 import java.util.Map;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.ExecutionException;
-import java.util.concurrent.ForkJoinPool;
 import java.util.concurrent.Executors;
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeoutException;
 import java.util.concurrent.Executor;
 import java.util.concurrent.atomic.AtomicInteger;
 import java.util.concurrent.atomic.AtomicReference;
 import java.util.function.Function;
 import java.util.function.Predicate;
+import java.util.function.Supplier;
 import java.util.stream.Collectors;
 import java.util.stream.IntStream;
 import lombok.val;
+import org.junit.After;
 import org.junit.Assert;
+import org.junit.Before;
 import org.junit.Test;
 
 /**
  * Unit tests for the Futures class.
  */
 public class FuturesTests {
+    private ScheduledExecutorService executor;
```

Comment:
Please change this class to extend from `ThreadPooledTestSuite` and override `getThreadPoolSize` to return 1.

That base class will take care of your executor.

While you're at it, there is at least one other test in this class that makes its own executor (and never shuts it down!). Please change that too to use `executorService()` instead.